courses/models.py

```diff
@@ -301,6 +308,24 @@ def first_unexpired_run(self):
             lambda course_run: course_run.is_unexpired,
         )
 
+    @property
+    def unexpired_run_dates(self):
+        """
+        Gets all the start dates for unexpired CourseRuns associated with this Course
+        """
+        return list(
+            map(
+                lambda unexpired_course_run: {
+                    "id": unexpired_course_run.pk,
+                    "date": unexpired_course_run.start_date,
+                },
+                filter(
+                    lambda course_run: course_run.is_unexpired,
+                    self.courseruns.all().order_by("start_date"),
+                ),
+            )
+        )
```

Comment:
This can be simplified. First, I don't see why we need to limit the fields returned. Why not just return `CourseRun` objects? If we do that, this function can be expressed like this

```python
import operator as op
#...
@property
def unexpired_run_dates(self):
  return filter(
    op.attrgetter("is_unexpired"),
    course.courseruns.all().order_by("start_date")
  )
```

Reply:
Done.